enarx-keep-sev/src/error.rs

```diff
@@ -0,0 +1,54 @@
+// SPDX-License-Identifier: Apache-2.0
+
+use std::error::Error;
+use std::fmt::Formatter;
+
+/// Describes the range of errors one might encounter.
+#[derive(Debug)]
+pub enum Kind {
+    /// Helpful for adding context messages around other errors.
+    Context(String),
+}
+
+impl std::fmt::Display for Kind {
+    fn fmt(&self, f: &mut Formatter) -> std::fmt::Result {
+        match self {
+            Kind::Context(msg) => write!(f, "{}", msg),
+        }
+    }
+}
+
+/// Quickly construct a Context KeepError.
+#[macro_export]
+macro_rules! context {
+    ($msg:expr, $err:expr) => {
+        KeepError::context($msg, std::boxed::Box::new($err))
+    };
+}
+
+/// The primary error type for the crate.
+#[derive(Debug)]
+pub struct KeepError {
```

Comment:
I would like to see `KeepError` impl `std::error::Error` with the `source()` method.